### PR TITLE
Move optional mocking behind flag

### PR DESCRIPTION
**Pull request by shanejearley**
Makes the basic setup unreliant on any externally-installed CLI tool and moves mocking behind a flag.

**Basic setup**
Run `npm run dev` from project root. Uses the AWS-deployed `dev` stage endpoints service endpoints by default in the Vue app, so no upfront config is necessary. This stage can be overridden in .env with STAGE variable.

**Mocking setup**
Install the SAM CLI on your machine and then run `npm run dev --mock`. Uses the local `dev` stage endpoints (which you may even prefer to override with `prod` to do things like reference production static resources (email templates, for example) and pass them into your local development for testing. SAM CLI can crawl our AWS CDK stacks to find and serve API Gateway endpoints. We pass in environment variables to the frontend development server (process.env.PUBLIC_MOCK, process.env.PUBLIC_{"api-name")_URL to switch to local endpoint URLs. 

Wrote more about the setup in the [README.md](https://github.com/consensusnetworks/casimir/tree/enhancement/config-defaults#setup). More about the SAM/CDK mocking integration [here](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-cdk-getting-started.html), and our script using the CLI is [here](https://github.com/consensusnetworks/casimir/blob/enhancement/config-defaults/scripts/cdk/mock).

Lastly just moved AWS-deployed API endpoints behind our custom domain for clean reference based on the current environment's stage.

### CODE REVIEW

**Comment by shanejearley** [comment]
> Makes the basic setup unreliant on any externally-installed CLI tool and moves mocking behind a flag.
> 
> **Basic setup** Run `npm run dev` from project root. Uses the AWS-deployed `dev` stage endpoints service endpoints by default in the Vue app, so no upfront config is necessary. This stage can be overridden in .env with STAGE variable.
> 
> **Mocking setup** Install the SAM CLI on your machine and then run `npm run dev --mock`. Uses the local `dev` stage endpoints (which you may even prefer to override with `prod` to do things like reference production static resources (email templates, for example) and pass them into your local development for testing. SAM CLI can crawl our AWS CDK stacks to find and serve API Gateway endpoints. We pass in environment variables to the frontend development server (process.env.PUBLIC_MOCK, process.env.PUBLIC_{"api-name")_URL to switch to local endpoint URLs.

Just to clarify, "we pass in environment variables" should be "the script passes environment variables" as a result of the `--mock` flag. No additional config is necessary, although you can optionally override the current profile, stage, etc. in your local `.env`.

> 
> Wrote more about the setup in the [README.md](https://github.com/consensusnetworks/casimir/tree/enhancement/config-defaults#setup). More about the SAM/CDK mocking integration [here](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-cdk-getting-started.html), and our script using the CLI is [here](https://github.com/consensusnetworks/casimir/blob/enhancement/config-defaults/scripts/cdk/mock).
> 
> Lastly just moved AWS-deployed API endpoints behind our custom domain for clean reference based on the current environment's stage.

I will wait for you @ccali11 to pull down and try/confirm/tear-apart before merging.

**Comment by ccali11** [comment]
- Last link in the comment on this PR is broken
- Should note which version of the AWS CLI we're using (1 or 2) in README
- In README, it may be worth letting developer know which backend resources / services are relevant (in this case, maybe not bc it appears it is just a signup form that will eventually be either replaced or outshined by other services / features)

Otherwise - dude - this is so clean and well done. Nothing to tear to shreds. I set it up and it worked perfectly for me, although I already had my aws user and SAM cli installed. May be worth having Hawyar take try to run it (purposely didn't tag him so that I don't bias his experience).

**Comment by ccali11** [comment]
Going to skim through your code now.

File: infrastructure/cdk/bin/index.ts [hunk 1]
```diff
@@ -1,14 +1,22 @@
 #!/usr/bin/env node
 import 'source-map-support/register'
 import * as cdk from 'aws-cdk-lib'
+import { pascalCase } from '@casimir/lib'
```
Comment (ccali11) [inline infrastructure/cdk/bin/index.ts:4]: How long did it take to familiarize yourself with the SAM/CDK? This all looks great.

File: infrastructure/cdk/lib/dns/dns-stack.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+     * DnsStack class constructor super method.
+     * 
+     * Shortest name:  {@link (DnsStack:constructor:super)}
+     * Full name:      {@link (DnsStack:constructor:super)}
+     */
+    super(scope, id, props)
+
+    const { project, stage } = props
+
+    const domain = 'casimir.co'
```
Comment (ccali11) [inline infrastructure/cdk/lib/dns/dns-stack.ts:41]: casimir.co let's go!

File: infrastructure/cdk/lib/website/website-stack.ts [hunk 3]
```diff
@@ -16,13 +24,16 @@ import { S3Origin } from 'aws-cdk-lib/aws-cloudfront-origins'
...
+  public readonly service: string = 'Website'
+  public readonly assetPath: string = '../../apps/website/dist'
+
   /**
    * WebsiteStack class constructor.
    * 
    * Shortest name:  {@link (WebsiteStack:constructor)}
    * Full name:      {@link (WebsiteStack:constructor)}
    */
-  constructor(scope: Construct, id: string, props?: StackProps) {
```
Comment (ccali11) [inline infrastructure/cdk/lib/website/website-stack.ts:25]: what's the difference between `WebsiteStackProps` and `StackProps`?

File: scripts/aws/configure [hunk 4]
```diff
@@ -1,40 +1,43 @@
-#!/bin/sh
```
Comment (ccali11) [inline scripts/aws/configure:1]: Did you write all of these scripts in the `scripts` directory from scratch? V impressive.

File: services/users/api/signup.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ import {
 } from '@aws-sdk/client-pinpoint'
 import { APIGatewayEvent } from 'aws-lambda'
 import { pascalCase } from '@casimir/lib'
```
Comment (ccali11) [inline services/users/api/signup.ts:7]: Where is this `@casimir/lib`?

File: services/users/api/signup.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@ import {
...
-import { SignupResponse } from '../env'
+import { APIGatewayResponse } from '../env'
```
Comment (ccali11) [inline services/users/api/signup.ts:8]: What's the need/rationale for this change?

File: services/users/index.ts [hunk 7]
```diff
@@ -1,20 +1,29 @@
...
-        'Access-Control-Allow-Origin': '*',
-        'Access-Control-Allow-Headers' : 'Content-Type',
-        'Access-Control-Allow-Methods': 'OPTIONS,POST,PUT,GET,DELETE',
-        'Content-Type': 'application/json'
-      },
-      statusCode: 200,
+      headers: defaultHeaders,
+      statusCode: 404,
       body: {
-        message: 'pong'
```
Comment (ccali11) [inline services/users/index.ts:17]: I liked 'pong'

**Review by ccali11 — APPROVED** [review]
So nicely done, @shanejearley. Feel free to ignore my questions if it's more trouble than it's worth answering them. Otherwise down to review whenever. Everything built nicely and was able to use the form and receive an email.

File: apps/website/src/main.ts [hunk 8]
```diff
@@ -6,6 +6,9 @@ import '@/index.css'
 import { createRouter, createWebHistory } from 'vue-router'
 import routes from '~pages'
 
+console.log('Creating app...', import.meta.env)
```
[the remark this answers is not in the document]
Comment (shanejearley) [reply apps/website/src/main.ts:9]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import.meta

**Pull request merged**